### ISSUE TITLE
FaucetNodeOptions are incorrect

### ISSUE THREAD

**Issue by pixelzoom** [phetsims/scenery-phet#752]
Discovered while converting scenery-phet demos to TypeScript. This is the first usage of FaucetNode in a TypeScript caller, so the first time that this problem has been exposed.

FaucetNode is:

```typescript
export default class FaucetNode extends AccessibleSlider( Node, 0 )
```

In demoFaucetNode.ts, attempting to pass options to FaucetNode complains about missing `valueProperty` and `enabledProperty`:

<img width="1036" alt="screenshot_1734" src="https://user-images.githubusercontent.com/3046552/176580624-067bbfed-2802-435b-9c80-d9c58bdff87d.png">

Those options should _not_ be part of the FaucetNodeOptions API. But they are being included via AccessibleSliderOptions, which gets them from AccessibleValueHandlerOptions.  

I believe that I converted FaucetNode  to TypeScript.  But I'm not familiar with the subtleties of making FacuetNode look like a slider by using the AccessibleSlider trait.  I'm guessing that there's some work to be done to omit those options from FaucetNodeOptions, and to (perhaps?) set their values in the options that are passed to `super(...)`.   
 
@jessegreenberg or @zepumph would you please take care of this? And remove the `ts-ignore` in demoFaucetNode.ts when you're done please.

**Comment by zepumph** [comment]
I can see 2 ways forward. The first is where we manually omit or convert valueProperty and enabledRangeProperty to optional each time we use an AccessibleValueHandler. This is most clear, but likely we will be got by this in the future because it is needed at each usage.

The other option is to make these optional arguments in the typescript options in AccessibleValueHandler, and then use runtime assertions to make sure they are provided. This would only touch common code, and not usages. Let's discuss!

**Comment by zepumph** [comment]
I discussed this with @pixelzoom and we like the way of the above commit. I went through other usages of `AccessibleSlider()` in typescript to omit the options. We are ready to close
